### ISSUE TITLE
Contested Claims | Attorney Fees | Refactor Claimant model into subclasses

### ISSUE THREAD

**Issue by nanotone** [department-of-veterans-affairs/caseflow#14502]
## Description
Split up the current Claimant model into VeteranClaimant and DependentClaimant subclasses, leaving any common code in a new Claimant base class if needed.

## Acceptance criteria
- [ ] New subclasses named VeteranClaimant and DependentClaimant
- [ ] Database migration for updating the `type` column to match each record's new class name

## Background/context/resources
The goal is to support claimants that fall into one of the following types:
- veteran
- dependent
- attorney
- unlisted (attorney)

as well as any additional types that might be required for future work on contested claimants. Currently the Claimant class supports the first two types, but as we support more types, the implementation of specific methods is likely to diverge. Single-table inheritance allows us to organize these divergent implementations while easily sharing common table columns (decision review, participant id, etc).

## Technical notes
Depends on issue #14380 and PR #14487 which adds the `type` column to the claimants table, enabling single-table inheritance. The migration from that first pass leaves every record with a type of `Claimant`.

The migration required for this ticket will need to determine, on a row-by-row basis, whether each existing record is a veteran or a dependent. The `payee_code` is not entirely reliable for this: a `00` value implies a veteran claimant, but claimants only need a payee code for VBMS cases. Joining with the 3 decision review tables gives access to `veteran_is_not_claimant`, and could be done in SQL or higher-level Rails code.
